pkg/istio-agent/health/health_check_test.go

```diff
@@ -124,7 +124,7 @@ func TestWorkloadHealthChecker_PerformApplicationHealthCheck(t *testing.T) {
 			{Healthy: true},
 			{Healthy: false},
 		}
-		httpHealthStatuses := [4]bool{true, false, true, false}
+		httpHealthStatuses := [4]bool{true, false, false, true}
```

Comment:
shouldn't this match `expectedHTTPEvents`?

Reply:
This test is confusing. Actually the 2 and 3 healthcheck result does not change, so the 3rd check does not result into status change and no callback at all.